include/soci/soci-platform.h

```diff
@@ -175,4 +175,23 @@ private: \
     #define SOCI_DUMMY_RETURN(x) return x
 #endif
 
+#define SOCI_OS_LINUX       0x0001
+#define SOCI_OS_FREE_BSD    0x0002
+#define SOCI_OS_APPLE       0x0003
+#define SOCI_OS_WINDOWS     0x0004
```

Comment:
`SOCI_OS` is (indirectly) used further down the line in preprocessor checks for enabling specific `exchange_traits` and `type_conversion`s. I don't know how this would play out with `enum class` and/or `constexpr`.